src/main/java/sparksoniq/spark/DataFrameUtils.java

```diff
@@ -225,24 +225,25 @@ public static String getGroupbyProjectSQL(
             }
             if (applyCount) {
                 queryColumnString.append("count(`");
+            } else if (applyDistinct) {
+                queryColumnString.append(serializerUdfName);
+                queryColumnString.append("(");
+                queryColumnString.append("first(");
             } else {
                 queryColumnString.append(serializerUdfName);
                 queryColumnString.append("(");
-                if (applyDistinct) {
-                    queryColumnString.append("array_distinct(");
-                }
```

Comment:
I don't think this needs to be extracted to a separate if condition as the part with serializerUDF and collect_list are still identical. 

I think it would still work if only the line with array_distinct was replaced with array(first(